---
title: "latgrad_share_v3"
output: html_document
editor_options: 
  chunk_output_type: console
---

lactobacillus analyses

supplementary lactobacillus check
simple experiment where lactobacillus at standard concentration (od600=0.1) was added to individual, frozen (dead) flies. DNA was then extracted using mechanical disruption and either +/- protK. 
```{r}
rm(list=ls())

require(phyloseq)
require(ggplot2)
require(tidyverse)

#setappropriate working directory
#setwd()

#import into phyloseq 
#import otu table and convert to matrix
otu_table<- read.csv("latgrad_lacto/otu_table.csv", sep=",", row.names=1)
otu_table<- as.matrix(otu_table)

#import taxonomy cleaned table and convert to matrix
taxonomy<- read.csv("latgrad_lacto/taxonomy_clean.csv", sep=",", row.names=1)
taxonomy<- as.matrix(taxonomy)

#read in metadata
metadata = read.table("latgrad_lacto/protk2_meta_clean.csv", sep=",", row.names=1, header=TRUE)

#import as phyloseq objects
OTU<- otu_table(otu_table, taxa_are_rows = TRUE)
TAX<- tax_table(taxonomy)
META<- sample_data(metadata)

#check for same number of OTU names across all three data types 
taxa_names(TAX)
taxa_names(OTU)

#check sample names, but order doesn't matter 
#if different number of sample names, phyloseq removes the unmatched ones 
sample_names(OTU)
sample_names(META)

#import into phyloseq object 
physeq<- phyloseq(OTU, TAX, META)

#call the phyloseq object to see the summary of data
physeq

#to look at the different data types in the phyloseq object
physeq@sam_data

#do basic investigation of distribution of reads by sample type
#want to see distribution of reads
size.df<- as.data.frame(sample_data(physeq))
#get sums
size.df$libsize<- sample_sums(physeq)

#visualize
#plot.readdepth<- 
ggplot(size.df, aes(libsize)) +
  geom_histogram(aes(fill=group), color="black", size=0.1, alpha=0.8, binwidth=1000)+ 
  scale_fill_manual(values =c("#66c2a4", "#8c96c6", "#df65b0", "#bdbdbd")) +
  theme_classic() +
  xlab("sequencing depth") +
  ylab("count")

#get numbers on this
size.df %>%
  group_by(group) %>%
  summarise(min=min(libsize), median = median(libsize), mean = mean(libsize), max=max(libsize))

#subsample down to 500. lost 14 samples. 
lacto.r <- rarefy_even_depth(physeq, sample.size=500, replace=FALSE, trimOTUs=TRUE, rngseed=11)

#only 4 flies but 15 fly + lacto 
data.frame(lacto.r@sam_data) %>%
  group_by(group, method) %>%
  summarise(n=n())

#make relative abundance plots
lacto.r.ra <- 
  lacto.r %>% 
  tax_glom(taxrank = "Family") %>%
  psmelt () %>%
  filter(group != "neg.ctrl") %>%
  group_by(base) %>%
  mutate(ind.sum = sum(Abundance)) %>%
  group_by(Family) %>%
  mutate(relabund = Abundance/ind.sum) %>%
  filter(relabund > 0.01) %>%
  #remove any family unassembled) 
  filter(Family != "f__")

#plot.lacto.rab<- 
ggplot(lacto.r.ra, aes(x=method, y=Abundance, fill=Family)) +
  geom_bar(stat="identity", position="fill") +
  scale_fill_manual(values =c("#bababa", "#ef8a62", "#e0e0e0", "#67a9cf")) +
  facet_grid(~group) +
  theme_classic() +
  theme(strip.background = element_blank()) +
  annotate("segment",x=Inf,xend=-Inf,y=Inf,yend=Inf,color="black",lwd=1) +
  xlab("") + 
  ylab("Relative abundance > 1%")

#ggsave(plot.lacto.rab, file="v2_plots/lactocheck_rab.pdf", height=3, width=5.5)

#also plot change in lacto between +/- protK 
#plot.lacto.protk<- 
lacto.r.ra %>%
  filter(group == "Fly + Lacto") %>%
  filter(Family == "f__Lactobacillaceae") %>%
    ggplot(., aes(x=method, y=relabund)) +
      geom_boxplot(color="#67a9cf") +
      geom_point(color="#67a9cf") + 
      theme_classic() + 
      xlab("") + 
      ylab("Lactobacillaceae RA") 

#ggsave(plot.lacto.protk, file="v2_plots/lactocheck_protk.pdf", height=3, width=2)

#stats
lacto.r.ra %>%
  filter(group == "Fly + Lacto") %>%
  filter(Family == "f__Lactobacillaceae") %>%
  t.test(data=., relabund~method)

```